stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/header/MessagesHeaderView.kt

```diff
@@ -0,0 +1,349 @@
+package io.getstream.chat.android.ui.messages.header
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.ColorStateList
+import android.content.res.TypedArray
+import android.graphics.Typeface
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.LayoutInflater
+import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.core.content.res.use
+import androidx.core.view.isVisible
+import com.getstream.sdk.chat.style.TextStyle
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamMessagesHeaderViewBinding
+import io.getstream.chat.android.ui.utils.extensions.getDimension
+
+public class MessagesHeaderView : ConstraintLayout {
```

Comment:
We've voted to name this class in the `Channel...` manner. However, there is already `ChannelsView` so `MessagesHeaderView` sounds less confusing for me.

Reply:
Oh, these are gonna be messy 😄 

I think our best chance of fixing up the naming at this point is taking a look at all the classes we have when we're done with the new design, and then renaming / resorting them into packages.